superset/views/core.py

```diff
@@ -703,6 +703,7 @@ def explore(  # pylint: disable=too-many-locals,too-many-return-statements
                 datasource_id, datasource_type, form_data
             )
         except SupersetException:
+            flash("The dataset associated with this chart no longer exists", "danger")
```

Comment:
I think there could be other reasons for an exception, we should use the actual error message

Reply:
And don't forget to add i18n: 

```python
_("Error occurred when trying to open the chart: %(error), error=utils.error_msg_from_exception(ex))
```

Reply:
Thanks for suggestions! I updated message. I think it would be worth to add an issue to refactor `datasource` <-> `dataset` naming convention. WDYT? @nytai @ktmud

Reply:
@kkucharc I think it's definitely worth creating a separate PR to change the naming for user-facing text and messages, not so sure about the code though.